packages/new-compiler/src/translation-server/translation-server.ts

```diff
@@ -747,10 +711,6 @@ export class TranslationServer {
     try {
       const parsedLocale = parseLocaleOrThrow(locale);
 
-      if (!this.translationService) {
-        throw new Error("Translation service not initialized");
-      }
```

Comment:
Quite dumb check, we make sure `TranslationService` is never null in constructor, since it's not something heavy  we should delay creating